### PR TITLE
Disable failing tests on windows

### PR DESCRIPTION
**Pull request by emillon**
This adds `(enabled_if)` to tests that do not pass on Windows today.
The resulting situation is that `dune runtest` works on Windows, and corresponding tests can be fixed on a longer schedule.

### CODE REVIEW

File: test/passing/dune.inc [hunk 1]
```diff
@@ -2015,12 +2015,14 @@
 
 (rule
  (deps .ocamlformat dir1/dir2/.ocamlformat dir1/dir2/print_config.ml)
+ (enabled_if (<> %{os_type} Win32))
  (action
    (with-outputs-to print_config.ml.output
      (run %{bin:ocamlformat} dir1/dir2/print_config.ml --print-config --config=max-iters=2 %{dep:print_config.ml}))))
```
Comment (Julow) [inline test/passing/dune.inc:2021]: Would this one work if it was passed like this ? `%{dep:dir1/dir2/print_config.ml}`

File: test/passing/dune.inc [hunk 2]
```diff
@@ -2489,22 +2491,26 @@
...
  (deps .ocamlformat )
+ (enabled_if (<> %{os_type} Win32))
  (action
    (with-outputs-to verbose1.ml.output
      (run %{bin:ocamlformat} --print-config --doc-comments=before --config=doc-comments=before %{dep:verbose1.ml}))))
 
 (rule
  (alias runtest)
+ (enabled_if (<> %{os_type} Win32))
  (action (diff verbose1.ml.ref verbose1.ml.output)))
```
Comment (Julow) [inline test/passing/dune.inc:2502]: What's wrong with verbose* tests ?

File: test/projects/dune [hunk 3]
```diff
@@ -25,6 +25,7 @@
 
 (rule
  (alias runtest)
+ (enabled_if (<> %{os_type} Win32))
  (action
   (diff outside_detected_project.expected outside_detected_project.output)))
```
Comment (Julow) [inline test/projects/dune:30]: Missing `enabled_if` on the rule for that file. Also, why is this one not working on Windows?

File: test/passing/gen/gen.ml [hunk 4]
```diff
@@ -77,9 +81,12 @@ let emit_test test_name setup =
   in
   let extra_deps = String.concat " " setup.extra_deps in
   let enabled_if_line =
-    match setup.minimum_ocaml_version with
-    | None -> ""
-    | Some v -> Printf.sprintf "\n (enabled_if (>= %%{ocaml_version} %s))" v
+    match setup.minimum_ocaml_version, setup.enabled_if with
```
Comment (Julow) [inline test/passing/gen/gen.ml:84]: There's 6 uses of `minimum-ocaml-version`. Replacing it with `enabled-if` may be a smaller patch and would remove code.

File: test/passing/gen/gen.ml [hunk 5]
```diff
@@ -7,7 +7,8 @@ type setup =
   ; mutable base_file: string option
   ; mutable extra_deps: string list
   ; mutable should_fail: bool
-  ; mutable minimum_ocaml_version: string option }
+  ; mutable minimum_ocaml_version: string option
+  ; mutable enabled_if: string option }
```
Comment (emillon) [inline test/passing/gen/gen.ml:11]: @gpetiot I had to add a more general mechanism of disabling tests. We can either:

- keep both `.minimum-ocaml-version` and `.enabled-if` files, where the former is a shortcut for a version check and there's an error if both files are provided
- just keep `.enabled-if` files and migrate the existing `.minimum-ocaml-version` files to this new format

What do you prefer?

Comment (gpetiot) [reply test/passing/gen/gen.ml:11]: I would prefer to have only one test, so only the `.enabled-if`

Comment (emillon) [reply test/passing/gen/gen.ml:10]: done!

File: test/projects/dune (hunk 3 above)
Comment (gpetiot) [reply test/projects/dune:42]: I added the clause on the related rule

File: test/passing/dune.inc (hunk 1 above)
Comment (gpetiot) [reply test/passing/dune.inc]: I modified this line, let's wait to see if it works on windows indeed

**Comment by emillon** [comment]
@gpetiot I just rebased, fixed the new failing tests under `failing/`, this is now passing on Windows.

**Comment by gpetiot** [comment]
> @gpetiot I just rebased, fixed the new failing tests under `failing/`, this is now passing on Windows.

Do we know why `print_config` and `verbose1/2` tests are failing on windows?

**Comment by emillon** [comment]
I think it's because they print paths, but the point of that PR is that we can investigate the reason of the failures later while keeping everything green.

**Review by gpetiot — APPROVED** [review]
Looks good to me, thanks! Feel free to merge when ready!

**Pull request merged**